bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieServer.java

```diff
@@ -248,11 +267,13 @@ public void run() {
                     Thread.currentThread().interrupt();
                 }
                 if (!isBookieRunning()) {
-                    shutdown();
-                    break;
+                    LOG.info("BookieDeathWatcher noticed the bookie is not running any more, exiting the watch loop!");
```

Comment:
just want to make sure that removal of "shutdown()" call here is intentional.

Reply:
yes it is intentional. because the uncaught exception handle registered in death watcher thread will handle all the shutdown logic.